packages/react-components/react-virtualizer/src/hooks/useMeasureList.ts

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+
+export interface IndexedResizeCallbackElement {
+  handleResize: () => void;
+}
+/**
+ * Provides a way of automating size in the virtualizer
+ * Returns
+ * `width` - element width ref (0 by default),
+ * `height` - element height ref (0 by default),
+ * `measureElementRef` - a ref function to be passed as `ref` to the element you want to measure
+ */
+export function useMeasureList<
+  TElement extends HTMLElement & IndexedResizeCallbackElement = HTMLElement & IndexedResizeCallbackElement,
+>(currentIndex: number, refLength: number, totalLength: number, defaultItemSize: number) {
+  const widthArray = React.useRef(new Array(totalLength).fill(defaultItemSize));
+  const heightArray = React.useRef(new Array(totalLength).fill(defaultItemSize));
+
+  const refArray = React.useRef<Array<TElement | undefined | null>>([]);
+  const { targetDocument } = useFluent();
+
+  // the handler for resize observer
+  const handleIndexUpdate = React.useCallback(
+    (index: number) => {
+      const boundClientRect = refArray.current[index]?.getBoundingClientRect();
```

Comment:
As logn as we aren't making any layout changes around this we should avoid thrashing. The main thing to avoid is doing something like calling this line, making layout updates and then hitting this again.